llvm/test/DebugInfo/X86/cas-friendly.ll

```diff
@@ -1,37 +1,33 @@
 ; RUN: llc -mtriple=x86_64-apple-macosx12.0.0 -filetype=obj %s -o %t 
 ; RUN: llvm-dwarfdump -debug-line --debug-info %t | FileCheck %s
 
-; CHECK: 0x{{[0-9a-z]+}}:   DW_TAG_subprogram
-; CHECK-NEXT:                DW_AT_low_pc ([[LOC1:0x[0-9a-z]+]])
-
-; CHECK: 0x{{[0-9a-z]+}}:   DW_TAG_subprogram
-; CHECK-NEXT:                DW_AT_low_pc ([[LOC2:0x[0-9a-z]+]])
-
-; CHECK: 0x{{[0-9a-z]+}}:   DW_TAG_subprogram
-; CHECK-NEXT:                DW_AT_low_pc ([[LOC3:0x[0-9a-z]+]])
-
-; This test is checking if the cas-friendly debug info splits the line table into multiple parts per function, that is why the checks below are checking for three seperate line tables for three distinct functions in this test
+; This test checks if the line tables are emitted in a cas-friendly manner, which means they emit end_sequences when a function's contribution to the line table has ended. This is done to be able to split a function's contribution to the line table into individual cas blocks.
 
 ; CHECK: debug_line[0x{{[0-9a-z]+}}]
 ; CHECK-NEXT: Line table prologue:
 
 ; CHECK: Address            Line   Column File   ISA Discriminator Flags
 ; CHECK-NEXT: ------------------ ------ ------ ------ --- ------------- -------------
-; CHECK-NEXT: [[LOC1]]      1      0      1   0             0  is_stmt
-
-; CHECK: debug_line[0x{{[0-9a-z]+}}]
-; CHECK-NEXT: Line table prologue:
-
-; CHECK: Address            Line   Column File   ISA Discriminator Flags
-; CHECK-NEXT: ------------------ ------ ------ ------ --- ------------- -------------
-; CHECK-NEXT: [[LOC2]]      5      0      1   0             0  is_stmt
-
-; CHECK: debug_line[0x{{[0-9a-z]+}}]
-; CHECK-NEXT: Line table prologue:
-
-; CHECK Address            Line   Column File   ISA Discriminator Flags
-; CHECK-NEXT ------------------ ------ ------ ------ --- ------------- -------------
-; CHECK-NEXT [[LOC3]]      9      0      1   0             0  is_stmt
+; CHECK-NEXT: 0x{{[0-9a-z]+}}      1      0      1   0             0  is_stmt
+; CHECK-NEXT: 0x{{[0-9a-z]+}}      2      9      1   0             0  is_stmt prologue_end
+; CHECK-NEXT: 0x{{[0-9a-z]+}}      2     10      1   0             0 
+; CHECK-NEXT: 0x{{[0-9a-z]+}}      2      2      1   0             0 
+; CHECK-NEXT: 0x{{[0-9a-z]+}}      2      2      1   0             0  end_sequence
+; CHECK-NEXT: 0x{{[0-9a-z]+}}      5      0      1   0             0  is_stmt
```

Comment:
Should we be checking the actual opcodes here to make sure the File actually gets reset?

Reply:
As we discussed in our 1-1, the end_sequence resets the registers of the state machine, that is all we need to look for. I will add a comment for it in the test